src/tools/convert.py

```
@@ -0,0 +1,220 @@
+        if isinstance(v, list):
+            list_to_xml(xml_root, k, v)
+        elif isinstance(v, dict):
+            dict_to_xml(xml_root, v, k)
+        else:
+            attr[k] = str(v)
+
+    _tag = convert_yaml_key_to_xml_tag(tag)
+    ET.SubElement(xml_root, _tag, attrib=attr)
+
+
+def list_to_xml(xml_root: ET.Element, tag: str, l: List[Dict[str, Union[List, str]]]) -> None:
+    '''
+    Add to an XML object a list of sub elements
+    '''
+    for element in l:
+        _tag = convert_yaml_key_to_xml_tag(tag)
+        sub = ET.SubElement(xml_root, _tag)
+        for k, v in element.items():
+            if isinstance(v, list):
+                list_to_xml(sub, k, v)
+            else:
+                # Special case
+                # The graphml attribute is not a tag in XML but a text element
+                if 'graphml' == k:
+                    sub.text = f'<![CDATA[{v}]]>'
+                else:
+                    sub.set(k, str(v))
+
+
+def save_xml(xml_root: ET.Element, filename: str):
+    '''
+    Write an XML element in a filename
+    '''
+    xmlstr = minidom.parseString(ET.tostring(xml_root)).toprettyxml(indent="   ")
+    with open(filename, "w", encoding='utf8') as f:
+        f.write(unescape(xmlstr, entities={'&quot;': '"'}))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Convert a XML file into a YAML file.')
+    parser.add_argument('operation', help='yaml2xml or xml2yaml')
+    parser.add_argument('filename', help='Filename to convert')
+    parser.add_argument('--output', help='Output filename', default=None, nargs='?')
+    args = parser.parse_args()
+
+    if args.operation == 'yaml2xml':
+        yaml_as_dict = get_yaml_from_filename(args.filename)
+        xml_root = create_xml_root(yaml_as_dict)
+        dict_to_xml(xml_root, yaml_as_dict)
+        filename_converted = get_output_filename(args, 'yaml', 'xml')
+        save_xml(xml_root, filename_converted)
+    else:
```

Let's explicitly check for 'xml2yaml' here, and return an error if the provided string here isn't one of the 2 supported operations. e.g. it'd be super-confusing to typo `yaml2xm`, and have it silently fall back to trying to do `xml2yaml`